packages/client/src/hooks/use-wallet.ts

```diff
@@ -138,18 +138,21 @@ export default function useWallet(): {
 
     const connectWalletConnect = async () => {
         await wcProvider.enable();
+        const walletObj: Wallet = {
+            account: wcProvider.accounts[0],
+            providerName: 'walletconnect',
+            provider: wcProvider,
+        };
 
         // If provider is different, sest to the WC wallet
         if (wallet.provider !== 'walletconnect') {
-            setWallet({
-                account: wcProvider.accounts[0],
-                providerName: 'walletconnect',
-                provider: wcProvider,
-            });
+            mixpanel.track('wallet:connect', walletObj);
+            setWallet(walletOb);
         }
     };
 
     const disconnectWallet = () => {
+        mixpanel.track('wallet:disconnect', {});
```

Comment:
might be worth tracking `wallet.provider` here, so we can track if some provider is having possible disconnection errors